interface/main/main_screen.php

```diff
@@ -136,14 +156,17 @@ function generate_html_end()
     }
     $userid = $_SESSION['authId'];
     $form_response = empty($_POST['form_response']) ? '' : $_POST['form_response'];
-
     if ($form_response) {
         // TOTP METHOD enabled if TOTP is visible in post request
+        if (empty($_POST['totp'])) {
+            $errormsg = xl("Please enter a valid code from the authenticator app");
+            $errortype = "TOTP";
+        }
```

Comment:
This code block does not make sense to me. Won't it throw an error when user is responding with U2F?